### PR TITLE
Configuration - Nuvla authorized-redirect-urls and minors

### PR DESCRIPTION
**Pull request by 0xbase12**
Fixes https://github.com/SixSq/tasklist/issues/1956

### CODE REVIEW

File: code/src/sixsq/nuvla/server/resources/configuration_nuvla.clj [hunk 1]
```diff
@@ -33,13 +34,31 @@ default values.
...
+  [redirect-url]
+  (not-any? (fn [authorized-redirect] (str/starts-with? redirect-url authorized-redirect)) *authorized-redirect-urls*))
+
+
+(defn throw-is-not-authorised-redirect-url
+  [redirect-url]
+  (if (and redirect-url
+             (seq *authorized-redirect-urls*)
+             (is-not-authorized-url? redirect-url))
+    (let [error-msg (str "server configuration do not authorize following rediect-url: " redirect-url)]
```
Comment (konstan) [inline code/src/sixsq/nuvla/server/resources/configuration_nuvla.clj:57]: I suggest extracting the error message as a var and reusing it in the unit test 
https://github.com/nuvla/api-server/pull/549/files#diff-c471681fa6083d06b7e0241fd4a806a813e24debabc66f6a98d53588bb8deb52R174

File: code/src/sixsq/nuvla/server/resources/session.clj [hunk 2]
```diff
@@ -267,6 +268,7 @@ status, a 'set-cookie' header, and a 'location' header with the created
   (try
     (let [authn-info (auth/current-authentication request)
           body       (if (u/is-form? headers) (u/convert-form :template form-params) body)
+          _          (-> body :redirect-url config-nuvla/throw-is-not-authorised-redirect-url)
```
Comment (konstan) [inline code/src/sixsq/nuvla/server/resources/session.clj:271]: I see there is a unit test for `group`, but the updated `session` here and `user` below are not tested.

File: code/src/sixsq/nuvla/server/resources/configuration_nuvla.clj [hunk 3]
```diff
@@ -55,22 +74,24 @@ default values.
...
+                (pricing-impl/set-impl! pricing-instance)
+                (pricing-impl/set-api-key! stripe-api-key)
+                (alter-var-root #'*stripe-api-key* (constantly stripe-api-key))
+                (when-let [stripe-client-id (or (-> config-instance-url
+                                                    crud/retrieve-by-id-as-admin
+                                                    :stripe-client-id)
+                                                (env/env :stripe-client-id))]
+                  (alter-var-root #'*stripe-client-id* (constantly stripe-client-id))))
+              (log/error "Stripe-api-key configured but no princing implementation found!"))))
+        (when-let [authorized-redirect-urls (:authorized-redirect-urls nuvla-config)]
```
Comment (konstan) [inline code/src/sixsq/nuvla/server/resources/configuration_nuvla.clj:93]: I suggest placing this outside of the wrapping try/catch for loading Strype config, and creating its own try/catch if needed.

**Review by konstan — CHANGES_REQUESTED** [review]

File: code/src/sixsq/nuvla/server/resources/session.clj (hunk 2 above)
Comment (0xbase12) [reply code/src/sixsq/nuvla/server/resources/session.clj:271]: The critical redirection is in group resource. user and session redirection are just to return error messages. This is why it must be tested for group. I will add a test for user and more for the authorize-url function

**Comment by konstan** [comment]
LGTM

**Pull request merged**